gcli/src/meta/executor.rs

```diff
@@ -0,0 +1,224 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::{anyhow, Result};
+use wasmi::{AsContextMut, Engine, Extern, Linker, Memory, MemoryType, Module, Store};
+
+const PAGE_STORAGE_PREFIX: [u8; 32] = *b"gcligcligcligcligcligcligcligcli";
+
+/// HostState for the WASM executor
+#[derive(Default)]
+pub struct HostState {
+    pub msg: Vec<u8>,
+    pub height: u64,
```

Comment:
The field is not used too. I guess they should be used in `gr_block_height` and `gr_block_timestamp`?

Reply:
actually I'm not pretty sure about these two functions since I don't see gmeta using them but they are required in the generated metadata WASM, so I just gave them empty implementation

Reply:
oh I got it 

https://github.com/gear-tech/gear/blob/master/examples/new-meta/state-v3/src/wasm.rs

Reply:
> oh I got it
> 
> https://github.com/gear-tech/gear/blob/master/examples/new-meta/state-v3/src/wasm.rs
> 
> `block_number` and `timestamp` require real execution env, however, the meta command in gcli only parse the data types of the state of programs, so we actually don't have to implement these two functions in the metadata executor for now

so maybe there is no need in calling it like "executor", introduce just "fn call_metadata()" and that is?

Reply:
https://github.com/gear-tech/gear/pull/3366/commits/ad94391ea95f1886ef5a0e563afb0e38455c6f62